server/sonar-server/src/main/java/org/sonar/server/computation/step/ValidateProjectStep.java

```diff
@@ -147,10 +147,10 @@ private void validateProjectKey(Optional<ComponentDto> baseProject, String rawPr
 
     private void validateAnalysisDate(Optional<ComponentDto> baseProject) {
       if (baseProject.isPresent()) {
-        SnapshotDto snapshotDto = dbClient.snapshotDao().selectLastSnapshotByComponentUuid(session, baseProject.get().uuid());
+        java.util.Optional<SnapshotDto> snapshotDto = dbClient.snapshotDao().selectLastSnapshotByRootComponentUuid(session, baseProject.get().uuid());
         long currentAnalysisDate = analysisMetadataHolder.getAnalysisDate();
-        Long lastAnalysisDate = snapshotDto != null ? snapshotDto.getCreatedAt() : null;
-        if (lastAnalysisDate != null && currentAnalysisDate <= snapshotDto.getCreatedAt()) {
+        Long lastAnalysisDate = snapshotDto.isPresent() ? snapshotDto.get().getCreatedAt() : null;
```

Comment:
FYI, you could also write `snapshotDto.map(SnapshotDto::getCreatedAt()).orElse(null);`

Avoids an inlined if statement, but I'm not sure if it is more efficient (I would say it isn't but I haven't looked into the optimization the compiler and then the JVM can do on such code)